src/components/tab-group/tab-group.component.ts

```diff
@@ -72,6 +76,9 @@ export default class SlTabGroup extends ShoelaceElement {
   /** Disables the scroll arrows that appear when tabs overflow. */
   @property({ attribute: 'no-scroll-controls', type: Boolean }) noScrollControls = false;
 
+  /** Hide scroll buttons when inactive. */
+  @property({ attribute: 'auto-hide-scroll-buttons', type: Boolean }) autoHideScrollButtons = false;
```

Comment:
I actually thinking this should be the default behavior. What do you think, @lindsaym-fa?

TL;DR – when tabs scroll, we show arrows to the left and right by default (unless `no-scroll-controls` is present). The arrows are always visible, but this PR hides them when scrolling that direction isn't possible.

Open questions:

- [ ] Should this be the default?
- [ ] Should the arrows be hidden entirely or styled to look disabled?

Reply:
Agreed that it should be the default behavior!

I'm open to hiding the arrows entirely, though there's one UX issue that's a dealbreaker for me on this point: when using keyboard controls to operate the scroll arrows, the focus position is lost when you reach either the beginning or the end of the tab group. We ought to move focus to the last tab when scrolled to the end and to the first tab when scrolled to the start.

Alternatively, we could disable the start and end arrows and allow them to retain focus when the beginning or end of the tab group is reached, respectively.

Reply:
Just to clarify your first point regarding moving the focus to the first or last tab when reaching the beginning or end of the tab group, respectively, would that mean that the tab focus would only wrap around to the first or last tab when the tab-group is not scrollable?

Reply:
# Focus
If user is operating scroll arrows with keyboard, reaches end, and we hide the arrow
* If we move to last tab when scrolled to end etc, when `activation` is `auto`, that will change which tab panel is shown.
* If we move focus to active tab: If active tab isn't visible and we give it focus and scroll to it, the arrow will become enabled/visible again, which is confusing - the user will never see the hidden/disabled state, he'll just reach the end and be bounced to the beginning.

The user was in the middle of scrolling, so we know he wants to scroll. I think if the arrow that was focused on disappears, a reasonable focus point would be on the opposite arrow button.

If the arrow becomes disabled, I still think the reasonable behaviour would be to go to the opposite arrow, for the same reason.

# Disabled vs hidden
Disabling and hiding are both reasonable. It happens to be that I need to hide it (and I'm fine with just having that functionality), but supporting both is also reasonable.

If we want to support both - that's a discussion of how that should be done, there are many options, IDK.
If we disable the tab, could that work with `:disabled`? So a shoelace user could do something like
```css
sl-tab-group::part(scroll-button):disabled {
  visibility:hidden;
}
```
But IDK how that works with accessibility. We'd need to provide a way to move focus to somewhere reasonable. Or, shoelace could have explicit support for  both options.

Thoughts?

Reply:
Very interesting. In accessibility scenarios, the arrow buttons are at best redundant and at worst a problem. So can we just mark them as `aria-hidden="true"`? In O'Hara's example, the clear button is technically useful - it's an easier way to clear. But here, the only purpose of the buttons is to scroll, and the keyboard user already has a direct way to do that.

That would also mean that, in terms of accessibility, shoelace would be agnostic to hiding the buttons vs disabling them; in either case, they are ignored by screen readers etc.

I had written about auto vs manual activation - looking at [the w3 docs](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_selection_follows_focus) (in context of tab group [here](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_selection_follows_focus)), there's a particular context in which each is appropriate. We should consider adding that guideline to the shoelace tab group docs.

Reply:
> But here, the only purpose of the buttons is to scroll, and the keyboard user already has a direct way to do that.
>
> That would also mean that, in terms of accessibility, shoelace would be agnostic to hiding the buttons vs disabling them; in either case, they are ignored by screen readers etc.

I'd be fine with this. What do you think, @lindsaym-fa?

Reply:
Yes! Removing the scroll arrows from the tab index completely resolves my initial concern. Better yet because curious keyboard users don't currently get any reward from using the scroll arrows — tabbing back to the tab group resets the scroll position, so any interactions with the scroll arrows are wasted.

We could offer the choice to either hide or disable the arrows, but in this case now where the arrows only benefit users with pointing devices, I'd venture that hiding the arrows offers better UX. Discerning disabled states can be tricky.

Reply:
Great! I updated it to prevent the scroll button arrows from receiving focus when using the keyboard to scroll.

Also, now the default behavior is to hide the scroll button when they're not clickable. I also added a `fixed-scroll-controls` attribute option in case a user wants to keep the buttons visible.